### PR TITLE
Fixes #35548 - Change content source and registered_through proxy

### PR DESCRIPTION
**Pull request by stejskalleos**
#### What are the changes introduced in this pull request?
Change content source should change `registered_through` proxy as well

#### Considerations taken when implementing this change?
See steps to reproduce in https://bugzilla.redhat.com/show_bug.cgi?id=2123975

#### What are the testing steps for this pull request?
See steps to reproduce in https://bugzilla.redhat.com/show_bug.cgi?id=2123975

### CODE REVIEW

File: app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb [hunk 1]
```diff
@@ -328,6 +328,10 @@ def change_content_source
         host.content_facet.content_view = content_view
         host.content_facet.content_source = content_source
 
+        if Katello.with_remote_execution?
+          host.subscription_facet.registered_through =  URI.parse(content_source.url).host
```
Comment (jeremylenz) [inline app/controllers/katello/api/v2/hosts_bulk_actions_controller.rb:332]:
```suggestion
          host.subscription_facet.registered_through = URI.parse(content_source.url).host
```

**Review by jeremylenz — COMMENTED** [review]
Thanks @stejskalleos, tested and works perfectly.

Would also like to see the `test_change_content_source` updated with an assertion describing this change.

**Comment by ianballou** [comment]
@adamruzicka 's comment here (https://github.com/Katello/katello/pull/10271#issuecomment-1255971489) is suggesting that this PR will break the REX pull provider.  If yggdrasil isn't updated before the smart proxy gets updated, then the next request to reconfigure yggdrasil on the host will fail becase it'll be attempting to connect via the wrong smart proxy

**Comment by ianballou** [comment]
I confirmed that this is going to be an issue for the pull provider. yggdrasil needs to be reconfigured before changing the registered-through smart proxy.  I wonder if we could change the registered-through proxy from the host with subscription-manager via a job...

**Review by ianballou — APPROVED** [review]
Worked for me with https://github.com/theforeman/foreman/pull/9435 !

**Comment by jeremylenz** [comment]
@stejskalleos Are you making any more changes here then?

**Comment by stejskalleos** [comment]
> Are you making any more changes here then?

Nope, it should be enough I hope, plus @ianballou confirmed that it works.

**Comment by jeremylenz** [comment]
Thanks @stejskalleos!

**Pull request merged**